File: jfingerprintmanager/src/main/java/com/jesusm/jfingerprintmanager/base/BaseFingerprintManager.java
package com.jesusm.jfingerprintmanager.base;

import android.support.annotation.StyleRes;
import android.support.v4.app.FragmentManager;

import com.jesusm.jfingerprintmanager.JFingerprintManager;
import com.jesusm.jfingerprintmanager.base.ui.FingerprintBaseDialogFragment;
import com.jesusm.jfingerprintmanager.base.ui.System;

public class BaseFingerprintManager {
    protected FingerprintAssetsManager fingerprintAssetsManager;
    private System system;
    @StyleRes
    private int authenticationDialogStyle;

    public BaseFingerprintManager(FingerprintAssetsManager fingerprintAssetsManager, System system) {
        this.fingerprintAssetsManager = fingerprintAssetsManager;
        this.system = system;
    }

    protected void showFingerprintDialog(FingerprintBaseDialogFragment.Builder builder,
                                       FragmentManager fragmentManager,
                                       JFingerprintManager.FingerprintBaseCallback callback) {
        builder.withCallback(callback)
                .withCustomStyle(authenticationDialogStyle)
                .withFingerprintHardwareInformation(fingerprintAssetsManager);

        system.addDialogInfo(builder, fragmentManager);
        system.showDialog();
    }

    public void setAuthenticationDialogStyle(@StyleRes int style) {
        this.authenticationDialogStyle = style;
    }
}

File: jfingerprintmanager/src/main/kotlin/com/jesusm/jfingerprintmanager/base/BaseFingerprintManager.kt
package com.jesusm.jfingerprintmanager.base

import android.support.v4.app.FragmentManager
import com.jesusm.jfingerprintmanager.JFingerprintManager
import com.jesusm.jfingerprintmanager.base.ui.FingerprintBaseDialogFragment.Builder
import com.jesusm.jfingerprintmanager.base.ui.System

abstract class BaseFingerprintManager(val fingerprintAssetsManager: FingerprintAssetsManager,
                                      private val system: System) {

    var authenticationDialogStyle: Int = -1

    fun showFingerprintDialog(builder: Builder<*, *>,
                              fragmentManager: FragmentManager,
                              callback: JFingerprintManager.FingerprintBaseCallback) {
        builder.withCallback(callback)
                .withCustomStyle(authenticationDialogStyle)
                .withFingerprintHardwareInformation(fingerprintAssetsManager)

        system.addDialogInfo(builder, fragmentManager)
        system.showDialog()
    }
}